### PR TITLE
Multitargeting with core

### PR DESCRIPTION
**Pull request by virzak**
Created multitargeted library and tests.

Need the latest daily build of .net core 3

### CODE REVIEW

File: Source/WriteableBitmapExBlitAlphaRepro.Wpf/WriteableBitmapExBlitAlphaRepro.Wpf.csproj [hunk 1]
```diff
@@ -1,120 +1,28 @@
...
-  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Release|AnyCPU' ">
-    <PlatformTarget>AnyCPU</PlatformTarget>
-    <DebugType>pdbonly</DebugType>
-    <Optimize>true</Optimize>
-    <OutputPath>bin\Release\</OutputPath>
-    <DefineConstants>TRACE</DefineConstants>
-    <ErrorReport>prompt</ErrorReport>
-    <WarningLevel>4</WarningLevel>
+
+  <!-- Remove when resolved https://github.com/dotnet/cli/issues/10379 -->
```
Comment (teichgraf) [inline Source/WriteableBitmapExBlitAlphaRepro.Wpf/WriteableBitmapExBlitAlphaRepro.Wpf.csproj]: Is this workaround still required? If so, do you think we should wait with merging this commit until the underlying bug is fixed? https://github.com/dotnet/cli/issues/10379

**Comment by teichgraf** [comment]
Thanks a lot @virzak  👍 

How did you create that PR? Did you open VS 2019 and changed the target for the projects? What tools are required for this? 

What is the benefit of targeting .NET Core 3? Better future maintainability?

File: Source/WriteableBitmapExBlitAlphaRepro.Wpf/WriteableBitmapExBlitAlphaRepro.Wpf.csproj (hunk 1 above)
Comment (virzak) [reply Source/WriteableBitmapExBlitAlphaRepro.Wpf/WriteableBitmapExBlitAlphaRepro.Wpf.csproj]: This only affects examples where the resources are linked, so really not a big deal in my opinion. Anyone using .net core whether for web or console app would run into the same issue

**Comment by virzak** [comment]
@teichgraf, yes I used VS 2019, but you can also use VS Code and apparently [VS 2017](https://www.youtube.com/watch?v=gIlfHtqgKk8)

Advanntages are listed [here](https://blogs.msdn.microsoft.com/dotnet/2018/05/07/net-core-3-and-support-for-windows-desktop-applications/). Currently all the new APIs are released on .NET Core and are omitted from .NET framework, which makes me think that they plan to discontinue .net framework in a few years. For example [File.ReadAll*Async](https://github.com/dotnet/corefx/issues/11220) of an API that's only available for core.

**Comment by virzak** [comment]
Oh yeah, this PR multitargets to .net40 and .netcoreapp3.0. This isn't a breaking change for anyone who would upgrade using nuget. The only inconvenience that might occur is when working with the source code of the library directly. But both VS 2019 and VS Code are free and provide debugging capabilities.

**Comment by teichgraf** [comment]
Makes sense @virzak and thanks. I will take an in-depth look and test it out next week or so when I have a little more time at my heads, so don't worry if I won't merge it in soon. I have it on my todo list.  :-)

**Comment by teichgraf** [comment]
Thanks a lot @virzak 
I confirmed with VS 2017 that it works with .NET Core preview 3 as well. 
Once .NET Core 3 is out of preview I will merge this into the master branch. For now it's in the NetCorePreview branch.

**Pull request merged**